File: src/ts/magma/app/compile/type/resolve/template/FunctionType.ts
import { Type } from "../../../../../../magma/app/compile/type/Type";
import { Iterable } from "../../../../../../magma/api/collect/list/Iterable";
import { TypeCompiler } from "../../../../../../magma/app/TypeCompiler";
export class FunctionType implements Type {
	args: Iterable<string>;
	returns: string;
	constructor (args: Iterable<string>, returns: string) {
		this.args = args;
		this.returns = returns;
	}
	isFunctional(): boolean {
		return true/*unknown*/;
	}
	isVar(): boolean {
		return false/*unknown*/;
	}
	generateBeforeName(): string {
		return ""/*string*/;
	}
	generateSimple(): string {
		return TypeCompiler.generateType(this)/*unknown*/;
	}
}

File: src/java/magma/app/compile/type/resolve/template/FunctionType.java
package magma.app.compile.type.resolve.template;

import magma.api.collect.list.Iterable;
import magma.app.TypeCompiler;
import magma.app.compile.type.Type;

public record FunctionType(Iterable<String> args, String returns) implements Type {
    @Override
    public boolean isFunctional() {
        return true;
    }

    @Override
    public boolean isVar() {
        return false;
    }

    @Override
    public String generateBeforeName() {
        return "";
    }

    @Override
    public String generateSimple() {
        return TypeCompiler.generateType(this);
    }
}
